docs/core/concepts/flows.md

```diff
@@ -192,3 +192,38 @@ state_handler(flow: Flow, old_state: State, new_state: State) -> State
 The handler is called anytime the flow's state changes, and receives the flow itself, the old state, and the new state. The state that the handler returns is used as the flow's new state.
 
 If multiple handlers are provided, they are called in sequence. Each one will receive the "true" `old_state` and the `new_state` generated by the previous handler.
+
+## Terminal State Handlers
+
+Flows allow users to provide custom logic for determining the final State of a Flow.
+
+Terminal state handlers must have the following signature:
+
+```python
+terminal_state_handler(flow: Flow, state: State, task_states: Dict[Task, State]) -> Optional[State]
+```
+
+`flow` is the current Flow
+`state` is the current state of the Flow
+`task_states` contains states for Flow Tasks
+
+An example use case would be iterating through return states and indicating which reference tasks that have failed.
```

Comment:
```suggestion
For example, you may want to determine which reference tasks that have failed and add them to your final state message.
```

Reply:
Note that this is particularly useful for Cloud Hooks (/ Automations) which report the state message. Now users can customize the success/failure message so their notifications include useful information about what has failed.